### DISCUSSION TITLE
Toast and snackbar after closing the app

### DISCUSSION THREAD

**Discussion by true-perfect-code** [CommunityToolkit/Maui#2326] [Q&A]
I couldn't find any examples of how to set toast/snackbar with future dates/times (e.g. similar to Plugin.LocalNotification)?

Is that even possible?

Scenario: e.g. you want to set a reminder that should only appear in 5 days. When the application is closed, Toast/Snackbar message appears from the system (Android, iOS or windows) and when the user clicks on it, my app opens. A key can be passed as a parameter, e.g. because you have already stored it when generating the Toast/Snackbar (usually an ID of a database entry, such as scheduling or similar). This is particularly important so that after opening the app (via toast message from the system) the relevant data record is automatically displayed (so that the user knows why the toast message appeared in the first place).

By the way, I am surprised that this was not the primary idea of Toast/Snackbar at CommunityToolkit.Maui, because I can display a message in dozens of different ways via any UI component as well as via HTML/CSSS (but I use MAUI Blazor). Now real toast message with triggered events and parameter passing is tricky and actually what you need in good apps.

Thanks a lot
pc

**Comment by VladislavAntonyuk** [comment]
This is not a responsibility of the Snackbar/toast. 
as you correctly mentioned you should use Notifications for future display. Furthermore Snackbar and toast can be only executed when app is active, while notifications can be used in background. This is an operation system limitation by design

**Comment by true-perfect-code** [comment]
It's a bad thing that there are still basic things in MAUI that can't be implemented, but are already possible with other frameworks.

I really wonder what motivated the CommunityToolkit team to implement this toast/snackbar functionality if you can't control the timing?

Do you at least think that CommunityToolkit team could do something in this direction?

Regards
pc

**Reply by bijington** [reply]
Vlad is part of the CommunityToolkit team and did the work on Toast and SnackBar. These are standard concepts implemented by Google and Vlad replicated it across the other platforms. 

Why can't you use the local notification plugin?

**Comment by true-perfect-code** [comment]
I would like to take this opportunity to thank Vlad and all the other team members for their excellent work.

There are two reasons why I am asking here:
1. plugin.LocalNotification only supports Android and iOS, but not Windows
2. i would always prefer a functionality from CommunityToolkit/Maui instead of choosing a Nuget for a functionality. 

Regards
pc

**Reply by VladislavAntonyuk** [reply]
Notifications could be part of essentials. Please create a separate discussion or API design how would you see the notifications API implemented. We can discuss it next month on our Monthly standup

**Comment by true-perfect-code** [comment]
I am happy to refer you to Plugin.LocalNotification, because they have given this a lot of thought.

![image](https://github.com/user-attachments/assets/bb9cb727-2c06-4a89-a21b-50be2b587a79)

For me personally (but I think also for all other developers) the part with scheduled appointments for displaying Toast/Snackbar would be crucial. This allows you to realize really great things in every possible application. Of course, it is also essential to trigger the display (or opening of the app) of a corresponding data record in the application when you click on the app's toast/snack bar after a few days.

I created a topic here: https://github.com/CommunityToolkit/Maui/discussions/2327

Regards
pc
